dongmin/기능개발.js

```diff
@@ -0,0 +1,26 @@
+function solution(progresses, speeds) {
+  const answer = [];
+  const MAX_PERCENTAGE = 100;
+  const remainDays = progresses.map((day, idx) =>
+    Math.ceil((MAX_PERCENTAGE - day) / speeds[idx])
+  );
+
+  while (remainDays.length) {
+    let first = remainDays[0];
+    let count = 1;
+
+    // count : 가장 앞에 있는 기능이 완료 될 때 까지 배포가 지연되는 기능의 수
+    while (first >= remainDays[count]) {
+      count += 1;
+    }
+
+    // count 만큼 데이터를 앞에서부터 제거
+    for (let i = 0; i < count; i += 1) {
+      remainDays.shift();
+    }
```

Comment:
제거하는 로직을 반복해서 `shift()` 메서드를 호출하는 것보다, `splice()`를 통해 제거하는 방법은 어떨까요?

```suggestion
    remainDays.splice(0, count);
```

Reply:
단순히 메서드 변경만 한다면 두 방식 다 요소 제거 후, 배열 재 정렬이라는 점에서는 큰 차이가 없을 듯 합니다!
대신 하나하나 제거하는 방식이 아닌 기능 개발이 끝난 시점에 완료된 기능을 한번에 제거하는 부분이라면 `splice`로 푸는 방식이 훨씬 좋을 것 같습니다!